source/core/src/main/com/csse3200/game/components/CombatStatsComponent.java

```diff
@@ -248,6 +252,15 @@ public void hit(CombatStatsComponent attacker) {
             float damageReduction = armor / (armor + 233.33f); //max damage reduction is 30% based on max armor(100)
             int newHealth = getHealth() - (int) ((attacker.getBaseAttack() + attacker.buff) * (1 - damageReduction));
             setHealth(newHealth);
+
```

Comment:
do entity.getname instead it will be simpler

Reply:
Done